diff --git a/configure.ac b/configure.ac
index f0ab1b8..18460e8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -62,8 +62,8 @@ if test `echo "$BOXTYPE" | cut -b 1-2` == "dm"; then
 	AC_DEFINE(DREAMBOX, 1, [Dreambox hardware])
 	AC_DEFINE(KEY_PLAY_ACTUALLY_IS_KEY_PLAYPAUSE, 1,[define when rc sends a KEY_PLAY event for its KEY_PLAYPAUSE key])
 fi
-if test "$BOXTYPE" == "vusolo2" -o "$BOXTYPE" == "vuduo2"; then
-	AC_DEFINE(VUPLUS_RC_WORKAROUND, 1, [workaround for problem with vusolo2/vuduo2 event/rc driver])
+if test "$BOXTYPE" == "vusolo2" -o "$BOXTYPE" == "vuduo2" -o "$BOXTYPE" == "vusolose"; then
+	AC_DEFINE(VUPLUS_RC_WORKAROUND, 1, [workaround for problem with vusolo2/vuduo2/vusolose event/rc driver])
 fi
 
 AC_ARG_WITH(textlcd,
diff --git a/data/rc_models/rc_models.cfg b/data/rc_models/rc_models.cfg
index 8dc4719..c09e3a6 100644
--- a/data/rc_models/rc_models.cfg
+++ b/data/rc_models/rc_models.cfg
@@ -42,3 +42,4 @@ vusolo   vu
 vuultimo vuultimo
 vuuno    vu
 xp1000   xp1000
+vusolose vu
