diff --git a/lib/python/Plugins/Extensions/HbbTV/plugin.py b/lib/python/Plugins/Extensions/HbbTV/plugin.py
index 22d5f69..57c9186 100644
--- a/lib/python/Plugins/Extensions/HbbTV/plugin.py
+++ b/lib/python/Plugins/Extensions/HbbTV/plugin.py
@@ -421,7 +421,7 @@ def start_youtubetv_main(session, **kwargs):


 def menu_start_youtube(menuid, **kwargs):
-	if menuid == "mainmenu":
+	if menuid == "mainmenu" and config.plugins.youtubetv.onMainMenu.getValue():
		return [(_("YouTube TV"), start_youtubetv_main, "youtube_tv", 46)]
	return []

@@ -444,6 +444,7 @@ def Plugins(**kwargs):
	l.append(PluginDescriptor(where=PluginDescriptor.WHERE_AUTOSTART, fnc=auto_start_main))
	l.append(PluginDescriptor(where=PluginDescriptor.WHERE_SESSIONSTART, needsRestart=True, fnc=session_start_main, weight=-10))
	l.append(PluginDescriptor(name=_("YouTube TV"), where=PluginDescriptor.WHERE_MENU, fnc=menu_start_youtube))
+	l.append(PluginDescriptor(name=_("YouTube TV"), where=PluginDescriptor.WHERE_EXTENSIONSMENU, fnc=start_youtubetv_main))
	l.append(PluginDescriptor(name=_("YouTube TV Settings"), where=PluginDescriptor.WHERE_PLUGINMENU, fnc=plugin_setting_youtube))
	l.append(PluginDescriptor(name=_("Browser Start/Stop"), where=PluginDescriptor.WHERE_EXTENSIONSMENU, needsRestart=True, fnc=extension_toggle_browser))
	l.append(PluginDescriptor(name=_("HbbTV Applications"), where=PluginDescriptor.WHERE_EXTENSIONSMENU, needsRestart=True, fnc=extension_start_application))
diff --git a/lib/python/Plugins/Extensions/HbbTV/youtube.py b/lib/python/Plugins/Extensions/HbbTV/youtube.py
index c2cc2ce..dd268c3 100644
--- a/lib/python/Plugins/Extensions/HbbTV/youtube.py
+++ b/lib/python/Plugins/Extensions/HbbTV/youtube.py
@@ -13,6 +13,7 @@ from __init__ import _
 config.plugins.youtubetv = ConfigSubsection()
 config.plugins.youtubetv.showhelp = ConfigYesNo(default = False)
 config.plugins.youtubetv.uri = ConfigText(default = "http://www.youtube.com/tv", visible_width = 50, fixed_size = False)
+config.plugins.youtubetv.onMainMenu = ConfigYesNo(default = False)
 
 vbcfg.g_youtubetv_cfg = config.plugins.youtubetv
 
@@ -164,8 +165,10 @@ class YoutubeTVSettings(ConfigListScreen, Screen):

		entryUri = getConfigListEntry(_("YouTube TV URL"), config.plugins.youtubetv.uri)
		entryShowHelp = getConfigListEntry(_("Do not show YouTube TV Starter again"), config.plugins.youtubetv.showhelp)
+		entryOnMainMenu = getConfigListEntry(_("Display YouTube in main menu"), config.plugins.youtubetv.onMainMenu)
		self.menulist.append(entryUri)
		self.menulist.append(entryShowHelp)
+		self.menulist.append(entryOnMainMenu)
		self["config"].list = self.menulist
		self["config"].l.setList(self.menulist)
 
